test/xkeyboard-config-test.py.in

```diff
@@ -79,36 +86,60 @@ def xkbcomp(r='evdev', m='pc105', l='us', v='', o=''):
         setxkbmap.stdout.close()
         stdout, stderr = xkbcomp.communicate()
         if xkbcomp.returncode != 0:
-            print('ERROR: Failed to compile: {}'.format(' '.join(args)))
+            outstr.append('ERROR: Failed to compile: {}'.format(' '.join(args)))
+            success = False
         if xkbcomp.returncode != 0 or verbose:
-            print(stdout.decode('utf-8'))
-            print(stderr.decode('utf-8'))
+            outstr.append(stdout.decode('utf-8'))
+            outstr.append(stderr.decode('utf-8'))
 
     # This catches setxkbmap errors.
     except subprocess.CalledProcessError as err:
-        print('ERROR: Failed to compile: {}'.format(' '.join(args)))
-        print(err.output.decode('utf-8'))
+        outstr.append('ERROR: Failed to compile: {}'.format(' '.join(args)))
+        outstr.append(err.output.decode('utf-8'))
+        success = False
+
+    return success, '\n'.join(outstr) if outstr else None
 
 
-def parse(root, tool):
+def parse(path):
```

Comment:
Maybe should change the function name now, since it no longer parses the keymaps? `collect`?

Reply:
it parses the `evdev.xml` file though, so I guess that counts? :)